components/event-feed-service/pkg/integration_test/feed_purge_test.go

```diff
@@ -182,9 +183,28 @@ func TestPurgeServer(t *testing.T) {
 				})
 				require.NoError(t, err)
 
+				startTime := time.Now()
 				_, err = testSuite.purgeClient.Run(context.Background(), &data_lifecycle.RunRequest{})
 				require.NoError(t, err)
 
+				// wait until the purge job is done
+				ticker := time.Tick(1 * time.Second)
+				select {
+				case <-ticker:
```

Comment:
i might be missing something...
isn't this always going to drop into the ticker branch